Restic backup (#249)
pulibrary/pul-it-handbook: services/restic_backup.md

3. Losing this password will make our backup inaccessible

## Store the file and password

The access keyfile, and password are required every time Restic communicates with your repository. To make it easier to work with your repository, create a shell script containing your credentials.

  1. To keep your credentials secure, using a text editor, create the example script in the user’s who will run the backup's home directory, and run all your Restic scripts as this user. The example uses the `postgres` user and the vim text editor.
       ```bash
       sudo su - postgres
       mkdir -p .restic
       vim .env.restic
       ```
       
     Copy and paste the json keyfile’s content and replace and with your own Object Storage account’s keyfile.

     ```file
     ### repository on google cloud
     export GOOGLE_APPLICATION_CREDENTIALS='/var/lib/postgresql/.restic/pul-gcdc-filename.json'

     export RESTIC_ARCHIVE_REPOSITORY='gs:postgres-version-backup:yourpath'

     export RESTIC_REPOSITORY=$RESTIC_ARCHIVE_REPOSITORY
     export RESTIC_PASSWORD_FILE='/var/lib/postgresql/.restic.pwd'
     ```

>      The `/var/lib/postgresql` path above will be `/home/pulsys` for mariadb

Reviewer: ```suggestion
     If you are backing up postgreSQL, use the path `/var/lib/postgresql`.
     If you are backing up mariadb, use the path `/home/pulsys`.
```

    
  2. Create a password file to hold your Restic password:
     ```bash
     sudo su - postgres
     vim ~/.restic.pwd
     ```
     Enter your Rustic password and save the file/
     ```file
     ~/.restic.pwd
     secretpassword # goes into lastpass and ansible vault
     ```
     
## Backup All Databases

For postgresql use the [postgresql](postgresql) scripts as a cronjob. 

  1. Copy all files above in your ~/.restic directory:
     ```bash
     sudo su - postgres
     mkdir -p ~/.restic/log
     ```
  2. Make all the scripts executable by the `postgres` user:
     ```bash
     chmod u+x ~/.restic/{common.sh,full_pg_backup.sh,pg_backup.sh,prune.sh}
     ```

  3. Set Up automated backups by creating a cron job for the `postgres` user with the following:
     ```bash
      sudo su - postgres
      crontab -e
      ```
     Add a line that points to the `full_pg_backup.sh` script

    ```file
    0 5 * * * /var/lib/postgresql/.restic/full_pg_backup.sh
    ```
    
For mariadb do the following:

>   1. Copy all the files under the [mariadb](mariadb) to the `pulsys` user

Reviewer: ```suggestion
  1. Copy all the files under the [mariadb](mariadb) to the `pulsys` user:
```

     ```bash
     mkdir -p ~/.restic/log
     ```
  2. Make all the scripts executable by the `pulsys` user:
     ```bash
     chmod u+x ~/.restic/maria_backup.sh
     ```
  3. Set Up automated backups by creating a cron job for the `pulsys` user with the following:
     ```bash
      crontab -e
      ```

>      Add a line that points to the `full_pg_backup.sh` script

Reviewer: ```suggestion
     Add a line that points to the `full_pg_backup.sh` script:
```

    ```file
    0 5 * * * /home/pulsys/.restic/maria_backup.sh
    ```

> ## Restore a backup

Reviewer: ```suggestion
## Restore from a backup
```
